### PR TITLE
chore: insert requstIP and receivedAt into the payload [PIPE-1134]

### PR DESCRIPTION
**Pull request by BonapartePC**
# Description

insert requstIP and receivedAt into the payload

# Linear
https://linear.app/rudderstack/issue/PIPE-1134/[rudder-server]-add-receivedat-and-ip-address-to-s3-logs
## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

## Summary by CodeRabbit

- **New Features**
  - Added `receivedAt` and `requestIP` fields to request payloads for enhanced tracking and logging.

- **Tests**
  - Introduced test cases to verify the addition of `receivedAt` and `requestIP` fields in request payloads.

### CODE REVIEW

**Review by Sidddddarth — APPROVED** [review]

File: gateway/gateway_test.go [hunk 1]
```diff
@@ -1536,6 +1538,52 @@ var _ = Describe("Gateway", func() {
...
+					RequestIP  string `json:"requestIP"`
+				} `json:"batch"`
+			}
+			err = json.Unmarshal(jobForm.jobs[0].EventPayload, &job)
+			Expect(err).To(BeNil())
+			Expect(job.Batch[0].ReceivedAt).To(ContainSubstring("2024-01-01T01:01:01.000000001Z"))
+			Expect(job.Batch[0].RequestIP).To(ContainSubstring("dummyIP"))
+		})
+
+		It("adds receivedAt in the request payload if it's not already present", func() {
```
Comment (lvrach) [inline gateway/gateway_test.go:1565]: You are missing a test case. If `receivedAt` is already present in the payload you should not override it.

File: gateway/gateway_test.go [hunk 2]
```diff
@@ -1536,6 +1538,52 @@ var _ = Describe("Gateway", func() {
...
+			req := &webRequestT{
+				reqType:        "batch",
+				authContext:    rCtxEnabled,
+				done:           make(chan<- string),
+				userIDHeader:   userIDHeader,
+				requestPayload: []byte(`{"batch": [{"type": "extract"}]}`),
+			}
+			req.ipAddr = "dummyIP"
```
Comment (lvrach) [inline gateway/gateway_test.go:1573]:
```suggestion
```

**Comment by lvrach** [comment]
please add linear ticket

**Review by lvrach — APPROVED** [review]

File: gateway/gateway_test.go (hunk 1 above)
Comment (BonapartePC) [reply gateway/gateway_test.go:1565]: I am already testing it in the above test

File: gateway/gateway_test.go (hunk 2 above)
Comment (BonapartePC) [reply gateway/gateway_test.go:1573]: Done

**Review by cisse21 — APPROVED** [review]

**Pull request merged**